pkg/cmd/lint-epic-issue-refs/lint_epic_issue_refs.go

```diff
@@ -0,0 +1,322 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"regexp"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+var (
+	fixIssueRefRE    = regexp.MustCompile(`(?im)(?:close[sd]?|fix(?:e[sd])?|resolve[sd]?):?\s+(?:(?:(#\d+)|([\w.-]+[/][\w.-]+#\d+))[,.;]?(?:[ \t]+|$))+`)
+	informIssueRefRE = regexp.MustCompile(`(?im)(?:see also|informs):?\s+(?:(?:(#\d+)|([\w.-]+[/][\w.-]+#\d+))[,.;]?(?:[ \t]+|$))+`)
```

Comment:
I figured out that we also want to support issue refs for Jira issues. The easiest way to do that is:

- Update these two regexes to also match Jira issue keys
- Add tests that include Jira issue refs
- Ensure Jira issue refs are handled correctly in the code that uses these regexes and the code that consumes the found issue refs.

I think the updated regexes would look like this:

```suggestion
	fixIssueRefRE    = regexp.MustCompile(`(?im)(?:close[sd]?|fix(?:e[sd])?|resolve[sd]?):?\s+(?:(?:(#\d+)|([\w.-]+[/][\w.-]+#\d+)|([A-Z]+-[0-9]+))[,.;]?(?:[ \t]+|$))+`)
	informIssueRefRE = regexp.MustCompile(`(?im)(?:see also|informs):?\s+(?:(?:(#\d+)|([\w.-]+[/][\w.-]+#\d+)|([A-Z]+-[0-9]+))[,.;]?(?:[ \t]+|$))+`)
```